Written by: s0dyy <maxime.sorin@clever-cloud.com>
Upstream: no
Reason: clang banned by distribution

---
 src/runtime/eng/native/build-commons.sh     | 2 +-
 src/runtime/src/coreclr/build-runtime.sh    | 4 ++--
 src/runtime/src/native/corehost/build.sh    | 2 +-
 src/runtime/src/native/libs/build-native.sh | 2 +-
 4 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/runtime/eng/native/build-commons.sh b/src/runtime/eng/native/build-commons.sh
index d9eb04cd8..2bd1a1dac 100755
--- a/src/runtime/eng/native/build-commons.sh
+++ b/src/runtime/eng/native/build-commons.sh
@@ -386,7 +386,7 @@ while :; do
             ;;

         gcc*|-gcc*)
-            __Compiler="$lowerI"
+            __Compiler="${CC}"
             ;;

         keepnativesymbols|-keepnativesymbols)
diff --git a/src/runtime/src/coreclr/build-runtime.sh b/src/runtime/src/coreclr/build-runtime.sh
index 06f79ec34..8dfb5e9ce 100755
--- a/src/runtime/src/coreclr/build-runtime.sh
+++ b/src/runtime/src/coreclr/build-runtime.sh
@@ -84,7 +84,7 @@ __BuildType=Debug
 __CodeCoverage=0

 # Set the various build properties here so that CMake and MSBuild can pick them up
-__Compiler=clang
+__Compiler=${CC}
 __CommonMSBuildArgs=
 __ConfigureOnly=0
 __CrossBuild=0
diff --git a/src/runtime/src/native/corehost/build.sh b/src/runtime/src/native/corehost/build.sh
index 6d112876e..d5df1a1c9 100755
--- a/src/runtime/src/native/corehost/build.sh
+++ b/src/runtime/src/native/corehost/build.sh
@@ -15,7 +15,7 @@ __TargetArch=x64
 __TargetOS=linux
 __BuildType=Debug
 __CMakeArgs=""
-__Compiler=clang
+__Compiler=${CC}
 __CrossBuild=0
 __PortableBuild=1
 __RootBinDir="$__RepoRootDir/artifacts"
diff --git a/src/runtime/src/native/libs/build-native.sh b/src/runtime/src/native/libs/build-native.sh
index 4d712eb0e..2fb355821 100755
--- a/src/runtime/src/native/libs/build-native.sh
+++ b/src/runtime/src/native/libs/build-native.sh
@@ -39,7 +39,7 @@ __TargetArch=x64
 __TargetOS=linux
 __BuildType=Debug
 __CMakeArgs=""
-__Compiler=clang
+__Compiler=${CC}
 __CrossBuild=0
 __PortableBuild=1
 __RootBinDir="$__RepoRootDir/artifacts"
--
2.52.0
